src/Components/Server/src/ComponentsHub.cs

```diff
@@ -48,15 +50,23 @@ internal CircuitHost CircuitHost
         /// <summary>
         /// Intended for framework use only. Applications should not call this method directly.
         /// </summary>
-        public override async Task OnDisconnectedAsync(Exception exception)
+        public override Task OnDisconnectedAsync(Exception exception)
         {
-            await CircuitHost.DisposeAsync();
+            var circuitHost = CircuitHost;
+            if (circuitHost == null)
+            {
+                return Task.CompletedTask;
+            }
+
+            CircuitHost = null;
+            _circuitRegistry.AddInactiveCircuit(circuitHost);
+            return circuitHost.OnConnectionDownAsync();
```

Comment:
It looks like there's the possibility of a threading issue here. You're adding the circuithost to the pool before tearing down the connection - which means that it could be initialized while it's being uninitialized. I think we'll need an async semaphore to resolve this.

Reply:
Do we need to handle exceptions here? What's the expectation if your `OnConnectionDownAsync` throws?

Reply:
Yeah, I wasn't entirely sure about how we should treat exceptions thrown from `CircuitHandler` methods. Treating them like filters in MVC and letting the exception bubble in the most common seems like a fairly straightforward approach, although it does mean user code would have to be very carefully written. Thoughts?

Reply:
Does it take done the connection?

Reply:
Yeah, it would.

Reply:
We log the errors now.